tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py

```diff
@@ -30,35 +30,39 @@ def __init__(self, marionette):
 
     def tap_wallpaper(self):
         self.marionette.find_element(*self._wallpaper_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
         from gaiatest.apps.wallpaper.app import Wallpaper
         wallpaper = Wallpaper(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == wallpaper.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == wallpaper.name)
         self.apps.switch_to_displayed_app()
         return wallpaper
 
     def tap_gallery(self):
         self.marionette.find_element(*self._gallery_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
```

Comment:
As above, find the element before the tap (ideally verify that it's displayed using a wait), and then wait for it to be not displayed after.